### PR TITLE
KeyVault - /rng response should always return a value on success

### PR DESCRIPTION
**Pull request by maorleger**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by yl-ms — APPROVED** [review]

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json:1292]: Pure response models typically don't have required parameters. I understand from internal conversations this may be causing generator to also potentially return `undefined`, but I suggest you file a generator bug to avoid this for response models.

That said, I don't see a problem taking this. It's just not common in this scenario.

**Review by heaths — APPROVED** [review]
This shouldn't actually be necessary (see comment for details), but I don't see harm in taking it.

@johanste ?

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json
Comment (yl-ms) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json:1292]: @maorleger May I ask whether this is necessary for the JS SDK support?

Comment (maorleger) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json:1292]: Thanks folks, appreciate the reviews! It's not strictly necessary, no. I also wasn't aware that it's not common (thanks @heaths!) - happy to close this PR and see if we can do something in the generator for JS, or in the convenience layer. I was just thinking that making the change here would ensure everyone gets it when generating.

But, definitely happy to defer to you if it's not the right thing to do - let me know

Comment (johanste) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json:1292]: Required does fill a need in response models. One reason that it is uncommon to specify it is that the openapi 2/swagger specification made some unfortunate decisions such stating that "`required` should not be used for `readOnly` properties" - which led to the assumption that `required` was meaningless when describing responses. This was fixed in openapi 3.

If the property (key/value) is guaranteed to be returned from the service, marking it as required makes perfect sense. If anything in the toolchain complains, please let me know.

Comment (heaths) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json:1292]: Thanks for the context / history.

**Comment by maorleger** [comment]
The errors I see are:
- Swagger BreakingChange: The new version has new required property 'value' that was not found in the old version. + The versions have not changed.
  - Should be fine since we're modifying 7.3 preview in-place right?
- Swagger ModelValidation is complaining about expirationTime, which is not related to this change
- Finally, Swagger SDK Breaking CHange Tracking failed:

```
❌azure-sdk-for-go - preview/keyvault/v7.2-preview/keyvault - v55.3.0
Diff with approved:
+	Package was removed
Diff with base:
+	Package was removed
```

It's my first change in this repo, so i'm not too familiar with what is acceptable - anything I need to do / wait on before merging this? 

Thanks in advance!

cc @heaths

**Comment by heaths** [comment]
Through 7.3, this has been the norm for previews but may change starting with 7.4 to uphold recommendations. @JeffreyRichter or @johanste can you approve the "breaking" change? /cc @lmazuel for merging with elevated privileges in the face of validation errors.

**Comment by JeffreyRichter** [comment]
A required property can't be added after a GA version of the service because clients using an earlier version never considered the property required. This change would break cuients when upgrading. Do I understand the change that is happenning here?

**Comment by maorleger** [comment]
@JeffreyRichter thanks for reviewing. This endpoint is new for 7.3-preview and does not exist in the 7.2 swagger so it was not defined in the GA swagger

**Comment by JeffreyRichter** [comment]
I see, OK I approved the PR

**Comment by heaths** [comment]
@lmazuel can you merge this now that Jeffrey has approved the "breaking" change? We've recommended to the service team to use feature branches for 7.4-preview so hopefully overriding becomes unnecessary.

That said, @JeffreyRichter, @johanste, @akning-ms: despite feature branches being configured like release branches (I believe), should the break change validation still run? Might be good to mitigate mistakes, but can anyone with merge permissions merge a PR with failed checks?

**Comment by maorleger** [comment]
@lmazuel friendly ping on this PR - is there anything you need from me before merging this?

**Comment by heaths** [comment]
@lmazuel @jhendrixMSFT can we get this merged? It's correcting the swagger for what the service does so our generated code is correct (should not be nullable).

**Pull request merged**
